build-support/bin/generate_lockfile.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+requirements_file=3rdparty/python/requirements.txt
+constraints_file=3rdparty/python/constraints.txt
+
+VENV=$(mktemp -d -t pants_lockfile_venv.XXXXX)
+virtualenv "${VENV}"
```

Comment:
I don't think we have dev directions to make sure this the `virtualenv` binary exists (I don't have it). It probably makes sense to use `build-support/virtualenv` or else something like `python3 -mvenv`